core/templates/pages/exploration-player-page/services/exploration-recommendations.service.spec.ts

```diff
@@ -0,0 +1,206 @@
+
+    spyOn(urlService, 'getUrlParams').and.returnValue({
+      story_id: STORY_ID,
+      node_id: NODE_ID
+    });
+
+    spyOn(contextService, 'getExplorationId').and.returnValue(EXPLORATION_ID);
+  });
+
+  describe('when used in the editor', () => {
+    beforeEach(() => {
+      spyOn(contextService, 'getPageContext').and.returnValue(
+        ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR
+      );
+    });
+
+    it('should initialize with editor context', () => {
+      expRecsService = TestBed.get(ExplorationRecommendationsService);
+      expect(expRecsService.isInEditorPage).toBeTrue();
+    });
+
+    describe('in the Preview tab', () => {
+      beforeEach(() => {
+        spyOn(contextService, 'getEditorTabContext').and.returnValue(
+          ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.PREVIEW
+        );
+      });
+
+      it('should initialize with editor preview context', () => {
+        expRecsService = TestBed.get(ExplorationRecommendationsService);
+        expect(expRecsService.isInEditorPage).toBeTrue();
+        expect(expRecsService.isInEditorPreviewMode).toBeTrue();
+      });
+    });
+  });
+
+  describe('when used outside of the editor', () => {
+    beforeEach(() => {
+      spyOn(contextService, 'getPageContext').and.returnValue(
+        ServicesConstants.PAGE_CONTEXT.OTHER
+      );
+    });
+
+    it('should not initialize with editor context', () => {
+      expRecsService = TestBed.get(ExplorationRecommendationsService);
+      expect(expRecsService.isInEditorPage).toBeFalse();
+    });
+  });
+
+  describe('getRecommendendedSummaryDicts', () => {
+    let expRecsBackendApiService: ExplorationRecommendationsBackendApiService;
+    const AUTHOR_REC_IDS = ['author_rec_1', 'author_rec_2'];
+
+    class MockExplorationSummary {
+      id: string;
+      constructor(id?: string) {
+        this.id = id;
+      }
+    }
+
+    let authorRecommendations: LearnerExplorationSummary[];
+    let systemRecommendations: LearnerExplorationSummary[];
+
+    beforeEach(() => {
+      systemRecommendations = [
+        new MockExplorationSummary('system_rec_1') as LearnerExplorationSummary,
+        new MockExplorationSummary('system_rec_2') as LearnerExplorationSummary
+      ];
+
+      authorRecommendations = [
+        new MockExplorationSummary('author_rec_1') as LearnerExplorationSummary,
+        new MockExplorationSummary('author_rec_2') as LearnerExplorationSummary
+      ];
+
+      expRecsBackendApiService = TestBed.get(
+        ExplorationRecommendationsBackendApiService
+      );
+
+      spyOn(
+        expRecsBackendApiService, 'getRecommendedSummaryDicts').and.callFake(
+        (_, includeSystemRecommendations: string) => {
+          return Promise.resolve(
+            includeSystemRecommendations === 'true' ?
+            systemRecommendations.concat(authorRecommendations) :
+            authorRecommendations);
+        });
+    });
+
+    describe('when used outside of the editor', () => {
```

Comment:
`when used in other page context`* (As the below are test is made in the context of `PAGE_CONTEXT.OTHER`)

Reply:
done